webapp/lib/platform_constants.dart

```diff
@@ -4,13 +4,17 @@ import 'dart:convert';
 import 'dart:html';
 
 String _constantsFilePath = 'assets/firebase_constants.json';
+String _latestCommitHashConfigFilePath = 'assets/latest_commit_hash.json';
```

Comment:
I don't think this should be in the platform part of the codebase, as platform is used here to refer to the firebase and cloud pubsub services. So I think it fits better for this file to be read directly in the view part of the app, and also without the controller intermediation as there's no user interaction about it.

Reply:
Ok. I've moved the logic into the view